File: debug_i18n.py
import json
import os

def load_json(path):
    with open(path, 'r') as f:
        return json.load(f)

base_path = '/Users/grandmaestro/Documents/P2PHub/frontend/src/locales/en/'
files = ['common.json', 'dashboard.json', 'marketing.json', 'academy.json', 'pro.json', 'social.json', 'cards.json', 'other.json']

merged = {}
for f in files:
    content = load_json(os.path.join(base_path, f))
    merged.update(content)

print("Root keys:", merged.keys())
if 'academy' in merged:
    print("Academy keys:", merged['academy'].keys())
else:
    print("Academy key NOT found at root!")

File: debug_i18n.js
const fs = require('fs');
const path = require('path');

const deepMerge = (target, source) => {
    const output = { ...target };
    if (source instanceof Object && target instanceof Object) {
        Object.keys(source).forEach(key => {
            if (source[key] instanceof Object) {
                if (!(key in target)) {
                    output[key] = source[key];
                } else {
                    output[key] = deepMerge(target[key], source[key]);
                }
            } else {
                output[key] = source[key];
            }
        });
    }
    return output;
};

const localesDir = path.join(__dirname, 'frontend/src/locales/en');
const files = [
    'common.json', 'dashboard.json', 'marketing.json', 'academy.json', 'pro.json', 'social.json', 'cards.json', 'other.json'
];

let en = {};
files.forEach(file => {
    const filePath = path.join(localesDir, file);
    if (fs.existsSync(filePath)) {
        const content = JSON.parse(fs.readFileSync(filePath, 'utf8'));
        en = deepMerge(en, content);
    }
});

console.log(JSON.stringify(en.network, null, 2));
